### ISSUE TITLE
ORA-01745 executing a SELECT * FROM WHERE column IN (large collection)

### ISSUE THREAD

**Issue by schlosna** [jOOQ/jOOQ#8203]
### Expected behavior and actual behavior: 

Expected behavior is that the following code executes without error and returns the relevant results.
```
    Collection<Long> ids = // collection larger than 65535 elements
    ctx.select(DSL.count()).from(TEST)
        .where(TEST.ID.in(ids))
        .fetchOne(0, int.class);
```

Actual behavior is that if one uses jooq to perform WHERE IN with a large collection (larger than 65535 elements) on Oracle, this results in `ORA-01745: invalid host/bind variable name`.

### Versions:

- jOOQ: 3.11.2 pro
- Java: 8u181
- Database (include vendor): Oracle 12.1
- OS: CentOS 7 Linux 
- JDBC Driver (include name if inofficial driver): ojdbc8 thin

**Comment by lukaseder** [comment]
Thank you very much for your report. The largest limitation we've encountered so far was PostgreSQL with its maximum of 32767 bind values (#5701). We've stopped looking for larger limits so far, but of course it makes sense to work around Oracle's limit as well.

As a quick workaround, you can inline all of your bind values by specifying `StatementType.STATIC_STATEMENT` for this particular statement, or by explicitly binding values in your in-list using `DSL.inline()`.

Of course, a much better workaround is to use an in-memory table and bind a single value, or to use a temp table, see also:
https://blog.jooq.org/2017/03/30/sql-in-predicate-with-in-list-or-with-array-which-is-faster/

**Comment by lukaseder** [comment]
Actually, the limit also seems to be 32768:
https://docs.oracle.com/en/database/oracle/oracle-database/18/lnpls/plsql-program-limits.html#GUID-00966B4C-B9A5-47D4-94AA-54AEBCC07CE9

**Comment by lukaseder** [comment]
Related new feature: #8206. The ability to explicitly set a threshold of the number of supported bind variables before inlining is enforced by jOOQ

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport scheduled for 3.11.10 (#8207) and 3.10.9 (#8208)

**Comment by schlosna** [comment]
Thanks @lukaseder , we'll test those options out. We were previously using hand maintained SQL joining against temp tables, and have been working to replace portions with jOOQ instead especially for read only operations to avoid the temp table insert & delete/truncate.
